src/common/function-macro/src/lib.rs

```diff
@@ -114,3 +116,93 @@ pub fn as_aggr_func_creator(_args: TokenStream, input: TokenStream) -> TokenStre
 pub fn range_fn(args: TokenStream, input: TokenStream) -> TokenStream {
     process_range_fn(args, input)
 }
+
+#[proc_macro_attribute]
+pub fn print_caller(args: TokenStream, input: TokenStream) -> TokenStream {
```

Comment:
request a brief document for this proc macro 🥲

Reply:
BTW, it looks like a util function is more suitable. User don't need to import `backtrace` manually, and function is easier to maintain

Reply:
Util function is itself introducing one more call stack frame. I feel this kind of tracking utilities is more naturally put in the form of attribute macros. Besides, `backtrace` is already imported in our codebase.

Reply:
>Util function is itself introducing one more call stack frame.

This doesn't matter, backtrace is a vector and we can skip 1 layer for the function.

> I feel this kind of tracking utilities is more naturally put in the form of attribute macros.

One problem with the macro way is we cannot decide where to capture and print the stack. It might be needed in a branch, an async task or someplace else. Thus I think a util function is more proper.

>Besides, backtrace is already imported in our codebase.

I'm not talking about adding new dependence. The point is every place that uses this macro should put `backtrace` into their toml and the rust file.

But I think this is not a blocking issue. We can resolve it later